internal/p2p/peermanager.go

```diff
@@ -867,6 +875,11 @@ func (m *PeerManager) Disconnected(ctx context.Context, peerID types.NodeID) {
 	delete(m.evicting, peerID)
 	delete(m.ready, peerID)
 
+	if peer, ok := m.store.Get(peerID); ok {
+		peer.LastDisconnected = time.Now()
```

Comment:
We'll need to start a routine that calls the dial waker after the cooldown. See the dial failed code as an example.